anesthetic/plotting/_matplotlib/hist.py

```diff
@@ -138,6 +142,20 @@ class Hist1dPlot(HistPlot):
     def _kind(self) -> Literal["hist_1d"]:
         return "hist_1d"
 
+    def _calculate_bins(self, data):
+        if range not in self.kwds:
+            q = self.kwds.get('q', 5)
+            q = quantile_plot_interval(q=q)
+            weights = self.kwds.get("weights", None)
+            xmin = quantile(data, q[0], weights)
+            xmax = quantile(data, q[-1], weights)
+            self.kwds["range"] = (xmin, xmax)
+            result = super()._calculate_bins(data)
+            self.kwds.pop("range")
+        else:
+            result = super()._calculate_bins(data)
+        return result
+
```

Comment:
In order to get 1d histogram binning and ranges to work, this code had to be copied from `anesthetic.plot.hist_plot_1d`. Without it, bins are calculated on the full range of the data.

This kind of thing would further justify moving our current `anesthetic.plot` code into `anesthetic.plotting._matplotlib`, but that would likely be a 3.0.0 move. Flagged in #318.

Reply:
Is there a test for this (that would fail on the master branch)? 
Since this is indeed in `hist_plot_1d` I can't quite see, why we need this...

Reply:
>Is there a test for this (that would fail on the master branch)?

Yes! [The new test](https://github.com/lukashergt/anesthetic/blob/50601e4169a2c5280b943cbd7e30a03bb43e9250/tests/test_samples.py#L1748-L1782) in the 'hist' branch will fail on master, since without this code levelled histograms adjust the ranges incorrectly.